compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -431,7 +435,11 @@ impl<'tcx> AsmCodegenMethods<'tcx> for CodegenCx<'_, 'tcx> {
                 }
             }
         }
-        if intel_syntax {
+
+        // Just to play it safe, reset the assembly syntax to att.
```

Comment:
Wouldn't this cause issues when `-Cllvm-args=-x86-asm-syntax=intel` is used? LLVM might still be trying to emit intel syntax asm for functions it compiles.

Reply:
LLVM doesn't emit asm as text, it emits the instructions directly and ignores the syntax. This only potentially affects inline asm in C code which would probably break anyways because most C code expects AT&T syntax for inline asm.

Reply:
> LLVM doesn't emit asm as text, it emits the instructions directly and ignores the syntax.

I think it would also break `--emit asm` producing valid assembly. Or does LLVM parse the asm blocks and then emit them as strings in the other format again?

> Well it works in the test? I believe what is really happening is that every asm block is isolated, given that LLVM parses the assembly anyway.

LLVM definitively merges global asm blocks (unfortunately). For example assembler macros leak between global asm blocks.

Reply:
The parsing at least seems to happen independently? Anyway this was already what we emitted, and the problem that got reported is that `.att_syntax` just got ignored, which this PR fixes.